pkg/kubelet/volumemanager/cache/desired_state_of_world.go

```diff
@@ -380,16 +463,22 @@ func (dsw *desiredStateOfWorld) UpdatePersistentVolumeSize(volumeName v1.UniqueV
 }
 
 func (dsw *desiredStateOfWorld) VolumeExists(
-	volumeName v1.UniqueVolumeName) bool {
+	volumeName v1.UniqueVolumeName, seLinuxMountContext string) bool {
 	dsw.RLock()
 	defer dsw.RUnlock()
 
-	_, volumeExists := dsw.volumesToMount[volumeName]
-	return volumeExists
+	vol, volumeExists := dsw.volumesToMount[volumeName]
+	if !volumeExists {
+		return false
+	}
+	if feature.DefaultFeatureGate.Enabled(features.SELinuxMountReadWriteOncePod) {
+		return vol.seLinuxFileLabel == seLinuxMountContext
```

Comment:
why does selinux matching impact volume existance?

Reply:
Because a volume with the same UniqueVolumeName, but a different SELinux label is *a different volume* from Kubelet point of view. Kubelet must unmount the it and mount it again with the right `-o context`.

I tried to add the SELinux context to `UniqueVolumeName`, but it is used by A/D controller and `node.status.volumesInUse` / `volumesAttached` and here we don't care about SELinux.

I tried to use another key for kubelet's ASW / DSW, but it got very messy quickly.

Reply:
Added a long comment